src/home_robot/home_robot/agent/goat_agent/goat_agent_module.py

```diff
@@ -0,0 +1,322 @@
+        blacklist_target=False,
+        matches=None,
+        confidence=None,
+        all_matches=None,
+        all_confidences=None,
+    ):
+        """Update maps and poses with a sequence of observations, and predict
+        high-level goals from map features.
+
+        Arguments:
+            seq_obs: sequence of frames containing (RGB, depth, segmentation, instance_segmentation)
+             of shape (batch_size, sequence_length, 3 + 1 + num_sem_categories,
+             frame_height, frame_width)
+            seq_pose_delta: sequence of delta in pose since last frame of shape
+             (batch_size, sequence_length, 3)
+            seq_dones: sequence of (batch_size, sequence_length) done flags that
+             indicate episode restarts
+            seq_update_global: sequence of (batch_size, sequence_length) binary
+             flags that indicate whether to update the global map and pose
+            seq_camera_poses: sequence of (batch_size, 4, 4) camera poses
+            init_local_map: initial local map before any updates of shape
+             (batch_size, 4 + num_sem_categories, M, M)
+            init_global_map: initial global map before any updates of shape
+             (batch_size, 4 + num_sem_categories, M * ds, M * ds)
+            init_local_pose: initial local pose before any updates of shape
+             (batch_size, 3)
+            init_global_pose: initial global pose before any updates of shape
+             (batch_size, 3)
+            init_lmb: initial local map boundaries of shape (batch_size, 4)
+            init_origins: initial local map origins of shape (batch_size, 3)
+            seq_object_goal_category: sequence of object goal categories of shape
+             (batch_size, sequence_length, 1)
+        Returns:
+            seq_goal_map: sequence of binary maps encoding goal(s) of shape
+             (batch_size, sequence_length, M, M)
+            seq_found_goal: binary variables to denote whether we found the object
+             goal category of shape (batch_size, sequence_length)
+            final_local_map: final local map after all updates of shape
+             (batch_size, 4 + num_sem_categories, M, M)
+            final_global_map: final global map after all updates of shape
+             (batch_size, 4 + num_sem_categories, M * ds, M * ds)
+            seq_local_pose: sequence of local poses of shape
+             (batch_size, sequence_length, 3)
+            seq_global_pose: sequence of global poses of shape
+             (batch_size, sequence_length, 3)
+            seq_lmb: sequence of local map boundaries of shape
+             (batch_size, sequence_length, 4)
+            seq_origins: sequence of local map origins of shape
+             (batch_size, sequence_length, 3)
+        """
+        # t0 = time.time()
+
+        # Reset the last channel of the local map each step when found_goal=False
+        if matches is not None or confidence is not None:
```

Comment:
Let's get rid of this "last channel of the local map" logic. See the comment about matching **instances in the current frame**.